frontend/pages/software/ManageSoftwarePage/components/ManageAutomationsModal/ManageAutomationsModal.tsx

```diff
@@ -71,15 +129,163 @@ const ManageAutomationsModal = ({
       ...newErrors,
     });
 
-    // URL validation only needed if software automation is checked
-    if (valid || !softwareAutomationsEnabled) {
-      onCreateWebhookSubmit({
-        destination_url,
-        enable_vulnerabilities_webhook: softwareAutomationsEnabled,
-      });
+    // Original config keys for software automation (webhook_settings, integrations)
+    const configSoftwareAutomations: ISoftwareAutomations = {
+      webhook_settings: {
+        vulnerabilities_webhook: {
+          destination_url,
+          enable_vulnerabilities_webhook: softwareVulnerabilityWebhookEnabled,
+        },
+      },
+      integrations: {
+        jira: integrations || [],
+      },
+    };
 
-      onReturnToApp();
+    if (!softwareAutomationsEnabled) {
+      // set enable_vulnerabilities_webhook to false and all jira.enable_software_vulnerabilities to false
+      configSoftwareAutomations.webhook_settings.vulnerabilities_webhook.enable_vulnerabilities_webhook = false;
+      const disableAllJira = configSoftwareAutomations.integrations.jira.map(
+        (integration) => {
+          return { ...integration, enable_software_vulnerabilities: false };
+        }
+      );
+      configSoftwareAutomations.integrations.jira = disableAllJira;
+    } else if (!jiraEnabled) {
+      if (valid) {
```

Comment:
I was confused what `valid` was. Renaming `validUrl` would help.

I also noticed the function populating this variable is called `validateWebhookURL`. Let's change the name of that function to `validateUrl` now that we're using it for different types of urls.

Reply:
Instead of doing another if/else here, you could start the function with this: 

```javascript
    } else if (!jiraEnabled) {
      if (!validUrl) {
        return;
      }
      // set enable_vulnerabilities_webhook to true and all jira.enable_software_vulnerabilities to false
...
```